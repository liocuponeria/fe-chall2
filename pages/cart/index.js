/* eslint-disable react/jsx-key */
import CartCard from '../../components/cart/cart-card'
import CartSummaryCard from '../../components/cart/cart-summary'
import styled from 'styled-components'

const Greeting = styled.h1`
    display: grid;
    grid-template-rows: 1fr;
`;

const FullCart = styled.div`
    display: grid;
    @media (max-width: 768px) {
        display: block;
      }
    grid-column-gap: 20px;
    grid-row-gap: 20px;
    
`;

const CartSummaryContainer = styled.div`
    grid-column-start: 3;
    grid-column-end: 4;
    grid-row: 1/2;
    grid-row-gap: 20px;
`;

const CartCardsContainer = styled.div`
    display: grid;
    grid-template-rows: 0.45fr 0.45fr 0.1fr;
    grid-column-start: 1;
    grid-column-end: 2;
    grid-row-gap: 20px;
    `;


function Cart({ cartProducts }) {
    return (
        <div>
            <Greeting>Meu carrinho</Greeting>
            <FullCart>
                <CartCardsContainer>
                    {cartProducts.map((product) => ( 
                        <CartCard key={product.id} {...product} />
                    ))}
                </CartCardsContainer>
                <CartSummaryContainer>
                    <CartSummaryCard />
                </CartSummaryContainer>
            </FullCart>
        </div>
    )
  }

  export async function getServerSideProps() {
    // Fetch data from external API
    const res = await fetch(`https://fakestoreapi.com/carts/5`)
    const cartFull = await res.json()
    const cart = cartFull.products
    let cartProducts = []
    for (let i = 0; i < cart.length ; i++) {
        let cartId = cart[i].productId
        let response = await fetch (`https://fakestoreapi.com/products/${cartId}`)
        let item = await response.json()
        item.quantity = cart[i].quantity
        cartProducts.push(item) 
    }
    // Pass data to the page via props
    return { props: { cartProducts } }
  }
  
  export default Cart