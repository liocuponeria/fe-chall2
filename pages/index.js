import styled from 'styled-components'
//components
import Card from '../components/card'

const Greeting = styled.h1`
    display: grid;
    grid-template-rows: 1fr;
`;
const CardContainer = styled.div`
    display: grid;
    grid-template-columns: 0.33fr 0.33fr 0.33fr;
    grid-column-gap: 20px;
    grid-row-gap: 20px;
    @media (max-width: 768px) {
      grid-template-columns: 1fr;
    }
    `;

function Home({ products }){
  return (
    <div>
      <Greeting>Sugestões para você</Greeting>
      <CardContainer>
        {products.map((product) => (
              <Card key={product.id} {...product}/> 
        ))}
      </CardContainer>
    </div>
  )
}

export async function getStaticProps() {
  // Calling the API endpoint to get products
  const res = await fetch('https://fakestoreapi.com/products')
  const products = await res.json()

  return {
    props: {
      products
    },
  }
}

export default Home