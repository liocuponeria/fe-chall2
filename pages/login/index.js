/* eslint-disable react/jsx-key */
import LoginForm from '../../components/login-form'
import styled from 'styled-components'

const Greeting = styled.h1`
    display: grid;
    grid-template-rows: 1fr;
    text-align: center;
`;
const LoginContainer = styled.div`
    display: grid;
    grid-column-gap: 20px;
    grid-row-gap: 20px;
    @media (max-width: 768px) {
        grid-template-columns: 1fr;
      }
    `;


function Login() {
    return (
        <div>
            <Greeting>Entrar</Greeting>
            <LoginContainer>
                    <LoginForm />
            </LoginContainer>
        </div>
    )
  }
  
  export default Login