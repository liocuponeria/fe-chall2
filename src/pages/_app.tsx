import type { AppProps } from 'next/app'
import AppProvider from 'hooks'
import { ThemeProvider } from 'styled-components'
import 'assets/fonts/css/fontello.css'
import { theme } from 'assets/theme'
import { GlobalStyle } from '../assets/theme'

const App: React.FC<AppProps> = ({ Component, pageProps }) => {
  return (
    <AppProvider>
      <GlobalStyle />
      <ThemeProvider theme={theme}>
        <Component {...pageProps} />
      </ThemeProvider>
    </AppProvider>
  )
}

export default App
