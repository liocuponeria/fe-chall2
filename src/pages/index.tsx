// import Head from "next/head";

import Background from '../components/Background'

const Home: React.FC = () => {
  return (
    <Background>
      <p>works</p>
    </Background>
  )
}

export default Home
