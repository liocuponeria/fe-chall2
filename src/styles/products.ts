import styled from 'styled-components';
import { shade } from 'polished';

export const ContainerProduct = styled.div`
  max-width: 1070px;
  width: 100%;
  margin: auto;
  justify-content: space-around;
  display: flex;
  flex-wrap: wrap;
  @media screen and (min-width: 320px) and (max-width: 663px) {
    justify-content: center;
  }
  @media screen and (min-width: 664px) and (max-width: 772px) {
    justify-content: space-around;
  }
  .product {
    margin-top: 28px;
    border: 1px solid #00000026;
    border-radius: 14px;
    display: block;
    color: red;
    align-items: center;
    background: var(--unnamed-color-ffffff) 0% 0% no-repeat padding-box;
    box-shadow: 0px 6px 12px #0000001f;
    border: 1px solid #00000026;
    opacity: 1;
    transition: transform 0.2s;
    &:hover {
      transform: translateX(2px);
    }
  }
  img {
    display: flex;
    margin: 3px auto;
    width: 328px;
    height: 203px;
    max-width: 90%;
    background: transparent
      url('img/Captura de Tela 2020-12-08 Ã s 15.59.48.png') 0% 0% no-repeat
      padding-box;
    border-radius: 14px 14px 0 0;
  }
  h4 {
    max-width: 300px;
    height: 40px;
    overflow: hidden;
    justify-content: center;
    display: flex;
    margin: 12px auto;
    font-family: 'Roboto';
    font-size: 16px;
    font-weight: 400;
    color: #000000b3;
    opacity: 1;
  }
  p {
    text-align: center;
    margin-top: 5px;
    font-size: 22px;
    color: #000000b3;
    font-family: 'Roboto';
    letter-spacing: 0px;
    opacity: 1;
    font-weight: 700;
  }
  button {
    justify-content: center;
    align-items: center;
    font-size: 16px;
    font-family: 'Roboto';
    font-weight: bold;
    width: 302px;
    height: 36px;
    background: transparent linear-gradient(90deg, #24C92B 0%, #099F0F 100%) 0% 0% no-repeat padding-box;
    border: 0;
    font-wei
    border-radius: 18px;
    opacity: 1;
    display: flex;
    margin: 19px 13px 27px 13px;
    color: #fff;
    border-radius: 180px;
    box-shadow: 0px 3px 6px #00000029;
    &:hover {
      background: ${shade(0.2, '#24c92b')};
    }
  }
`;