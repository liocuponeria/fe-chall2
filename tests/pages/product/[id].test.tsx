import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import ProductPage from 'pages/product/[id]'
import { theme } from 'pages/_app'
import { mockProducts } from '../../mock/products'

jest.mock('next/dist/client/router', () => ({
  useRouter: jest.fn().mockImplementation(() => ({
    query: {
      category: 'electronics',
    },
  })),
}))

describe('Product page', () => {
  it('should match the snapshot', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <ProductPage {...mockProducts[0]} />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
