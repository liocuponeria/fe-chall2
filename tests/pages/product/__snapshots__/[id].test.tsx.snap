// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product page should match the snapshot 1`] = `
<div
  className="Background__BackgroundContainer-sc-1ig5col-0 caskQE"
>
  <div
    className="Header__HeaderContainer-sc-1a8uskq-0 iAAsmi"
  >
    <div>
      <img
        alt="Cuponeria Logo"
        src="/imgs/logo.svg"
      />
    </div>
  </div>
  <div
    className="NavBar__NavBarContainer-sc-8wct3k-0 jtmBiI"
  >
    <div>
      <div
        className="ButtonNavBar__ButtonNavBarContainer-sc-109tumo-0 eVfgCa"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <i
          className="icon-inicio"
        />
        <span>
          início
        </span>
      </div>
      <div
        className="ButtonNavBar__ButtonNavBarContainer-sc-109tumo-0 eVfgCa"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <i
          className="icon-roupas"
        />
        <span>
          roupas
        </span>
      </div>
      <div
        className="ButtonNavBar__ButtonNavBarContainer-sc-109tumo-0 hmJHDC"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <i
          className="icon-eletronicos"
        />
        <span>
          eletrônicos
        </span>
      </div>
      <div
        className="ButtonNavBar__ButtonNavBarContainer-sc-109tumo-0 eVfgCa"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <i
          className="icon-acessorios"
        />
        <span>
          acessórios
        </span>
      </div>
    </div>
  </div>
  <div
    className="childrenMargin"
  >
    <div
      className="ProductDetails__ProductContainer-sc-15wgsak-0 iRTlRp"
    >
      <div
        className="prodDetailsImg"
      >
        <img
          alt=""
          src="https://fakestoreapi.com/img/71-3HjGNDUL._AC_SY879._SX._UX._SY._UY_.jpg"
        />
      </div>
      <div
        className="prodDetailsInfo"
      >
        <span
          className="prodDetailsTitle"
        >
          Mens Casual Premium Slim Fit T-Shirts 
        </span>
        <span
          className="prodDetailsPrice"
        >
          R$ 
          22.30
        </span>
        <div>
          <span
            className="prodDetailsDescription"
          >
            Slim-fitting style, contrast raglan long sleeve, three-button henley placket, light weight & soft fabric for breathable and comfortable wearing. And Solid stitched shirts with round neck made for durability and a great fit for casual fashion wear and diehard baseball fans. The Henley style round neckline includes a three-button placket.
          </span>
        </div>
        <div
          className="prodDetailsButtonContainer"
        >
          <a
            className="BuyButton__BuyButtonContainer-sc-1fz6kqp-0 XtTgO"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            comprar
          </a>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="Footer__FooterContainer-qgk6jt-0 gXkdUn"
  >
    <div>
      <i
        className="icon-facebook"
      />
      <i
        className="icon-instagram"
      />
      <i
        className="icon-twitter"
      />
      <i
        className="icon-youtube"
      />
    </div>
  </footer>
</div>
`;
